### PR TITLE
[FIX] account: check statement balance at each step

### PR DESCRIPTION
**Pull request by FlorianGilbert**
Previously, the check to see if a statement ending balance is the same as the computed balance was only done when moving from draft to posted which is wrong as people can modify the balance of posted statement programmatically which is the case when using online synchronization for example. Now perform that check at each change of state: draft->posted and posted->processed. Also the indication in the view to indicate that the balance was not the same was only displayed in draft, we removed that constraints as user should always see if there is a problem.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by csnauwaert** [comment]
can you rephrase the commit message to this? Thanks :)

[FIX] account: check statement balance at each step

Previously, the check to see if a statement ending balance is the same as the computed balance was only done when moving from draft to posted which is wrong as people can modify the balance of posted statement programmatically which is the case when using online synchronization for example. Now perform that check at each change of state: draft->posted and posted->processed. Also the indication in the view to indicate that the balance was not the same was only displayed in draft, we removed that constraints as user should always see if there is a problem.

**Comment by FlorianGilbert** [comment]
@csnauwaert Done and runbot green :)

**Comment by csnauwaert** [comment]
@robodoo r+
